test.rs

```diff
@@ -11,11 +11,38 @@ use ipc::{self, IpcOneShotServer, IpcReceiver, IpcReceiverSet, IpcSender, IpcSha
 use ipc::{OpaqueIpcSender};
 use router::ROUTER;
 use libc;
+use std::io::Error;
 use std::iter;
+use std::ptr;
 use std::sync::Arc;
 use std::sync::mpsc::{self, Sender};
 use std::thread;
 
+///XXXjdm Windows' libc doesn't include fork.
+#[cfg(not(windows))]
+// I'm not actually sure invoking this is indeed unsafe -- but better safe than sorry...
+pub unsafe fn fork<F: FnOnce()>(child_func: F) -> libc::pid_t {
```

Comment:
For the record: I actually think so too; but because of language limitations, this can currently be only achieved by using the Void crate -- which @pcwalton didn't want.